app/js/common/services/presenceReport.service.js

```
@@ -0,0 +1,121 @@
+var presenceReportResource = require('./../resources/presenceReport.resource.js');
+var reportTimeResource = require('./../resources/reportTime.resource.js');
+
+module.exports = angular
+.module('presenceReport.service', [
+  presenceReportResource.name,
+  reportTimeResource.name
+  ])
+.factory('presenceReportService', presenceReportService);
+
+presenceReportService.$inject = ['presenceReportResource',
+                                 'reportTimeResource',
+                                 'currentGroupDay',
+                                 '$mdToast'];
+
+function presenceReportService(
+  presenceReportResource,
+  reportTimeResource,
+  currentGroupDay,
+  $mdToast) {
+  var service = {
+    getPresenceReports: getPresenceReports,
+    getReportTimes: getReportTimes,
+    addReportTime: addReportTime,
+    deleteReportTime: deleteReportTime,
+    updateReportTime: updateReportTime,
+    updateEndTime: updateEndTime,
+    toggleErrorMsg: toggleErrorMsg
+  };
+  return service;
+
+  function getPresenceReports() {
+    return presenceReportResource.query({group_id: currentGroupDay.group_id,
+                                      day: currentGroupDay.day})
+    .$promise.then(function(presenceReports) {
+      return presenceReports;
+    }, function() {
```

i think that you should return an error messages